src/qt/walletcontroller.cpp

```diff
@@ -137,11 +135,24 @@ WalletModel* WalletController::getOrCreateWallet(std::unique_ptr<interfaces::Wal
     connect(wallet_model, &WalletModel::coinsSent, this, &WalletController::coinsSent);
 
     // Notify walletAdded signal on the GUI thread.
-    Q_EMIT walletAdded(wallet_model);
-
+    if (QThread::currentThread() == thread()) {
+        addWallet(wallet_model);
+    } else {
+        // Handler callback runs in a different thread so fix wallet model thread affinity.
+        wallet_model->moveToThread(thread());
+        bool invoked = QMetaObject::invokeMethod(this, "addWallet", Qt::QueuedConnection, Q_ARG(WalletModel*, wallet_model));
+        assert(invoked);
+    }
     return wallet_model;
 }
 
+void WalletController::addWallet(WalletModel* wallet_model)
+{
+    // Take ownership of the wallet model and register it.
```

Comment:
s/register/notify?

Reply:
comment must be updated,too (setparent & emit), ownership already done in if-clause and register was in  m_wallets.push_back(wallet_model) - if u update more u could do it.